src/image_ops.rs

```diff
@@ -1582,6 +1584,134 @@ Images are first converted to grayscale. Thresholds should be `>=0` and `< 256`.
         Example::new(1, 1, "thresh 120")
 );
 
+//-----------------------------------------------------------------------------
+// Threshold
+//-----------------------------------------------------------------------------
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+struct Tile(u32, u32);
+
+impl ImageOp for Tile {
+    fn apply(&self, stack: &mut ImageStack) {
+        let mut image = stack.pop();
+
+        let tiles = tile(&mut image, self.0, self.1);
+
+        for t in tiles.into_iter().rev() {
+            stack.push(t)
+        }
+    }
+
+    // takes 1, returns n
+    fn signature(&self) -> Option<(usize, usize)> {
+        None
+    }
+}
+
+fn tile(mut image: &mut DynamicImage, width: u32, height: u32) -> Vec<DynamicImage> {
+    let max_elem = (image.width() / width + 1) * (image.height() / height + 1);
```

Comment:
`((image.width() - 1) / width + 1) * ((image.height() - 1) / height + 1)` removes the possible overestimate here, not that it particularly matters. That obviously then requires image width and height to be strictly positive, but that's fine. It's probably worth adding an `assert!` that image width and height and tile width and height are all strictly positive at the start of this function anyway.